bindings/matrix-sdk-ffi/src/client_builder.rs

```diff
@@ -0,0 +1,87 @@
+use std::{fs, path::PathBuf, sync::Arc};
+
+use anyhow::Context;
+use matrix_sdk::{
+    ruma::UserId, store::make_store_config, Client as MatrixClient,
+    ClientBuilder as MatrixClientBuilder,
+};
+use sanitize_filename_reader_friendly::sanitize;
+
+use super::{client::Client, ClientState, RUNTIME};
+
+#[derive(Clone)]
+pub struct ClientBuilder {
+    base_path: Option<String>,
+    username: Option<String>,
+    homeserver_url: Option<String>,
+    inner: MatrixClientBuilder,
+}
+
+impl ClientBuilder {
+    pub fn new() -> Self {
+        Self {
+            base_path: None,
+            username: None,
+            homeserver_url: None,
+            inner: MatrixClient::builder().user_agent("rust-sdk-ios"),
+        }
+    }
+
```

Comment:
Docs should probably say that base path _must_ be set. Which let's me wonder if we shouldn't have that as a parameters on `new` so it is enforced and never `Option` that could fail at runtime (when calling `build`) . Wdyt?

Reply:
I think Andy actually wanted to have it as a `new` parameter, and I wanted to have it as a builder method instead. My reasoning for that was that it's the usual pattern for builders, and a `new ClientBuilder("/some/path")` invocation wouldn't be quite clear. (and it would be a very easy to catch programmer error when it isn't set)

However, this point only applies to Kotlin since UniFFI's Swift codegen requires callers to re-state argument names (as in `new ClientBuilder(base_path: "/some/path")`). So I don't insist on keeping it this way.

Reply:
Yea I generally favor API that makes invalid stated impossible to represent, which would require passing them to `new`. This sounds like an example of clash between SDK and client-side consistency. Happy to follow up on this in a new PR.